prboom2/src/m_random.c

```diff
@@ -146,6 +147,45 @@ void M_ClearRandom (void)
   rng.prndindex = rng.rndindex = 0;    // clear two compatibility indices
 }
 
+// [XA] Common random formulas used by codepointers
+
+//
+// P_RandomHitscanAngle
+// Outputs a random angle between (-spread, spread), as an int ('cause it can be negative).
+//   spread: Maximum angle (degrees, in fixed point -- not BAM!)
+//
+int P_RandomHitscanAngle(pr_class_t pr_class, fixed_t spread)
+{
+  int t;
+  int_64_t spread_bam;
+
+  // FixedToAngle doesn't work for negative numbers,
+  // so for convenience take just the absolute value.
+  spread_bam = (spread < 0 ? FixedToAngle(-spread) : FixedToAngle(spread));
+  t = P_Random(pr_mbf21);
+  return (int)((spread_bam * (t - P_Random(pr_mbf21))) >> 8);
```

Comment:
Is `8` a magic number or is there a defined constant we could put here?

Reply:
`>> 8` here is just a nicer `/ 255`, since we're effectively generating a random number between -255 and 255 and I wanna scale it to (-1,1). I guess there's no harm in defining a `RANDBITS` constant or something though.

Reply:
On second thought, maybe it actually should just be `/ 255`, since the value can be negative, and a right shift on that might be UB? lemme googlify it

Reply:
Replaced it with `/ 255` -- still just using the number instead of a constant, but eh.

Reply:
Sounds fine to me